src/pymorize/fesom_2p1/regridding.py

```diff
@@ -0,0 +1,370 @@
+        data_interpolated = np.ma.sum(w * data[inds], axis=1) / np.ma.sum(w, axis=1)
+        data_interpolated.shape = lons.shape
+        data_interpolated = np.ma.masked_invalid(data_interpolated)
+        return data_interpolated
+
+    elif how == "linear":
+        for qhull_path in qhull_paths:
+            if os.path.isfile(qhull_path):
+                logging.info(
+                    "Note: using precalculated file from {}".format(qhull_path)
+                )
+                try:
+                    qh = joblib.load(qhull_path)
+                    loaded_qhull = True
+                    break
+                except PermissionError:
+                    # who knows, something didn't work. Try the next path:
+                    continue
+        if not loaded_qhull:
+            points = np.vstack((mesh.x2, mesh.y2)).T
+            qh = qhull.Delaunay(points)
+            if dumpfile:
+                for qhull_path in qhull_paths:
+                    try:
+                        joblib.dump(qh, qhull_path)
+                        break
+                    except PermissionError:
+                        continue
+        data_interpolated = LinearNDInterpolator(qh, data)((lons, lats))
+        data_interpolated = np.ma.masked_invalid(data_interpolated)
+        return data_interpolated
+
+    elif how == "cubic":
+        for qhull_path in qhull_paths:
+            if os.path.isfile(qhull_path):
+                logging.info(
+                    "Note: using precalculated file from {}".format(qhull_path)
+                )
+                logging.info(
+                    "Note: using precalculated file from {}".format(qhull_path)
+                )
+                try:
+                    qh = joblib.load(qhull_path)
+                    loaded_qhull = True
+                    break
+                except PermissionError:
+                    # who knows, something didn't work. Try the next path:
+                    continue
+        if not loaded_qhull:
+            points = np.vstack((mesh.x2, mesh.y2)).T
+            qh = qhull.Delaunay(points)
+            if dumpfile:
+                for qhull_path in qhull_paths:
+                    try:
+                        joblib.dump(qh, qhull_path)
+                        break
+                    except PermissionError:
+                        continue
+        data_interpolated = CloughTocher2DInterpolator(qh, data)((lons, lats))
+    else:
+        raise ValueError("Interpolation method is not supported")
+
+
+##############################################################################
+
+
+def attach_mesh_to_rule(data, rule):
+    rule.mesh = load_mesh(rule.mesh_path)
+    return data
+
+
+def regrid_to_regular(data, rule):
+    mesh = load_mesh(rule.mesh_path)
+    box = rule.get("box", "-180, 180, -90, 90")
+    x_min, x_max, y_min, y_max = map(float, box.split(","))
+    x = np.linspace(x_min, x_max, int(x_max - x_min))
```

Comment:
Where is the regular grid resolution defined and how does the user define it in the config yaml?

Reply:
This is a bit clunky, I agree. By default it does 1˚x1˚, globally. I think it might be better to instead support something like CDO has, so `r360x180` for 1˚x1˚ or `r720x360` for 0.5˚x0.5˚.

Reply:
So with the current implementation I could simply set `box: "-360, 360, -180, 180"` to get a 0.5 degree resolution? It is not very pythonic but if this is what the users are used to fair enough. I would have expected to simply define a lat and lon resolution instead.

Reply:
I believe so, yes. Basically, the justification for this way of writing it down is "it does the same thing as pyfesom2"